frontend/packages/dev-console/src/components/monitoring/alerts/SilenceDurationDropdown.tsx

```diff
@@ -48,18 +60,42 @@ const SilenceDurationDropDown: React.FC<SilenceDurationDropDownProps> = ({ rule
         matchers,
         comment: '',
       };
-
-      coFetchJSON.post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload);
+      setSilencing(true);
+      coFetchJSON
+        .post(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`, payload)
+        .then(() => {
+          // eslint-disable-next-line promise/no-nesting
+          coFetchJSON(`${ALERT_MANAGER_TENANCY_BASE_PATH}/api/v2/silences`)
+            .then((silences) => {
+              rule.silencedBy = _.filter(
```

Comment:
`rule` is a prop received by the component. ideally we should avoid modification of props. we could use a deep copy of the prop instead. WDYT?